compiler/noirc_evaluator/src/ssa/opt/constant_folding.rs

```diff
@@ -226,9 +270,12 @@ impl<'brillig> Context<'brillig> {
         let mut side_effects_enabled_var =
             function.dfg.make_constant(FieldElement::one(), Type::bool());
 
+        let mut dom = DominatorTree::with_function(function);
```

Comment:
I moved it from `self` because of borrow checking. `self.get_constraint_map` requires `&mut self` and then the lookup of the value in `SimplificationCache::get` requires `&mut DominatorTree` which would be another borrow of `&mut self`. I tried by making the `DominatorTree` use a `RefCell` to no avail.

Why is it slowing down, it's just created once I think.

Reply:
Ah it's per block. Then it should be moved further up, and passed to `fold_constants_in_block` 🤦

Reply:
Moved up.